core/src/test/java/com/linecorp/armeria/client/ClientBuilderTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for {@link ClientBuilder}.
+ */
+class ClientBuilderTest {
+
+    @Test
+    void nonePlusSchemeProvided() {
+        final HttpClient client = new ClientBuilder("none+https://google.com/").build(HttpClient.class);
+        assertThat(client.uri().toString()).isEqualTo("https://google.com/");
+    }
+
+    @Test
+    void nonePlusSchemeUriToUrl() {
+        final HttpClient client = new ClientBuilder("none+https://google.com/").build(HttpClient.class);
+        assertThatCode(() -> client.uri().toURL()).doesNotThrowAnyException();
```

Comment:
How about using `.isEqualTo(URL)` to compare the actual value? Since we're asserting anyways doesn't hurt to have a stricter assertion of the behavior.

Reply:
Alternatively, I guess comparing to a known `URI` in the other test would implicitly check this and we wouldn't need it.

Reply:
I have made the change.

This UT was included, as prior to this PR uri().toURL() will fail with MalformedURLException. It's to avoid any regression check, once this PR is merged.

Reply:
Yeah - I think it's too lenient of a test though since it's easy to pass with a broken implementation that just returns a const `URI.create("github.com")` or something.

We're more concerned that the actual `URI` returned by Armeria is correct, while exception semantics of `toURL()` are implementation details of JDK, not Armeria. The other tests seem to take care of this, though I think it's even clearer without using `toString()` in them.

Reply:
Ah sorry I may have been confused by your comment - with the change it's probably fine :)